src/messages.rs

```diff
@@ -22,6 +32,64 @@ pub trait MessageTrait {
 pub struct Message {
     pub header: Header,
     pub payload: Payload,
+
+    pub metadata: TransportData,
```

Comment:
This should be optional (`None` for "own" generated messages)

Reply:
Done.

Reply:
My intention was to have

```diff
pub struct Message {
    pub header: Header,
    pub payload: Payload,

-   pub metadata: TransportData,
+   pub metadata: Option<TransportData>,
}
```

But again, we can take care of this during the next refactor